presto-main/src/main/java/com/facebook/presto/execution/scheduler/SimpleNodeSelector.java

```diff
@@ -55,6 +55,8 @@
     private final int maxSplitsPerNode;
     private final int maxPendingSplitsPerTask;
 
+    private final AtomicReference<ResettableRandomizedIterator<InternalNode>> randomCandidates = new AtomicReference<>(null);
```

Comment:
A `ResettableRandomizedIterator` is basically just a `List` of the originally data: https://github.com/prestodb/presto/blob/2e29401e90ab444a94ed9f3a836c6d980928dec3/presto-main/src/main/java/com/facebook/presto/execution/scheduler/ResettableRandomizedIterator.java#L23-L27

Should we just keep a `List` here? In that case we can just use `NodeSelector#selectRandomNodes ` . @arhimondr ? 

I remember this discussion happens before, let me see if I can find the previous discussion.